frontend/components/DocumentShowcase.js

```diff
@@ -0,0 +1,55 @@
+import React, {useEffect, useContext, useState} from "react";
+import ShowcaseItem from "./ShowcaseItem";
+import {TimelineArrow} from "./TimelineArrow";
+import {TimelineContext} from "../contexts/TimelineContext";
+import * as PropTypes from "prop-types";
+
+const DocumentShowcase = ({documents}) => {
+    const context = useContext(TimelineContext);
+    const [intervalDocuments, setIntervalDocuments] = useState([]);
+
+    useEffect(() => {
+        filterDocumentsByInterval();
+    }, [context.intervalSelected.start, context.intervalSelected.end]);
+
+    /**
+     * Filters the documents to only show the documents within the years in the selected timeineline
+     * interval with the starting year being inclusive and the ending year being exclusive
+     * (e.g. 1910-1915 interval would show documents from Jan. 1, 1910 to Dec. 31, 1914)
+     */
+    const filterDocumentsByInterval = () => {
+        const timelineStartDate = new Date(context.intervalSelected.start, 0);
+        const timelineEndDate = new Date(context.intervalSelected.end, 0);
+        const filteredDocs = documents.filter(doc => {
+            const docDate = new Date(doc.date);
+            return docDate >= timelineStartDate && docDate < timelineEndDate;
+        });
+        setIntervalDocuments(filteredDocs);
+    };
+
+    return (
+        <div id="showcase">
+            <TimelineArrow isLeft={true}/>
+            <div id="document-showcase">
+                {intervalDocuments.length > 0 
+                    ? intervalDocuments.map((document, index) =>
+                        <ShowcaseItem
+                            key={`${document.toString()}_${index}`}
+                            value={document}
```

Comment:
why not just use `index` as key? Is there a special reason for using `${document.toString()}_${index}` ?

Reply:
The React docs [1] they "don’t recommend using indexes for keys if the order of items may change" because may cause component state and performance issues. We pair the `toString()` with `index` to be able to distinguish between documents that might from the same publication or with the same title (e.g. Washington Bee, For Rent, For Sale).


[1] https://reactjs.org/docs/lists-and-keys.html#keys

Reply:
Aha, I see! I did not think about the order changing. Good point!